app/src/androidTest/java/com/radixdlt/android/fragment/MoreOptionsFragmentTest.kt

```diff
@@ -0,0 +1,74 @@
+package com.radixdlt.android.fragment
+
+import androidx.test.espresso.intent.Intents.intended
+import androidx.test.espresso.intent.matcher.IntentMatchers.toPackage
+import androidx.test.espresso.intent.rule.IntentsTestRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.MediumTest
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.rule.ActivityTestRule
+import androidx.test.uiautomator.UiDevice
+import com.radixdlt.android.R
+import com.radixdlt.android.activity.CreateNewWalletTest
+import com.radixdlt.android.ui.activity.MainActivity
+import com.radixdlt.android.ui.activity.NewWalletActivity
+import com.schibsted.spain.barista.assertion.BaristaVisibilityAssertions.assertDisplayed
+import com.schibsted.spain.barista.interaction.BaristaClickInteractions.clickOn
+import com.schibsted.spain.barista.rule.cleardata.ClearDatabaseRule
+import com.schibsted.spain.barista.rule.cleardata.ClearFilesRule
+import com.schibsted.spain.barista.rule.cleardata.ClearPreferencesRule
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.concurrent.TimeUnit
+
+/**
+ * Instrumented test, which will execute on an Android device.
+ *
+ * See [testing documentation](http://d.android.com/tools/testing).
+ */
+@RunWith(AndroidJUnit4::class)
+@MediumTest
+class MoreOptionsFragmentTest {
+
+    /**
+     * [ActivityTestRule] is a JUnit [@Rule][Rule] to launch your activity under test.
+     *
+     * Rules are interceptors which are executed for each test method and are important building
+     * blocks of Junit tests.
+     */
+    @get:Rule var newWalletActivityTestRule: ActivityTestRule<NewWalletActivity> =
+        ActivityTestRule(NewWalletActivity::class.java)
+
+    // Clear all app's SharedPreferences
+    @get:Rule
+    var clearPreferencesRule = ClearPreferencesRule()
+
+    // Delete all tables from all the app's SQLite Databases
+    @get:Rule var clearDatabaseRule = ClearDatabaseRule()
+
+    // Delete all files in getFilesDir() and getCacheDir()
+    @get:Rule
+    var clearFilesRule = ClearFilesRule()
+
+    @get:Rule
+    val intentsTestRule = IntentsTestRule(MainActivity::class.java)
+
+    @Test
+    fun testOpenReportIssueWebView() {
+        CreateNewWalletTest.createNewWallet()
+
+        clickOn(R.id.navigation_more_options)
+        assertDisplayed(R.string.more_options_fragment_xml_report_an_issue)
+
+        clickOn(R.id.reportAnIssueTextView)
+
+        Thread.sleep(TimeUnit.SECONDS.toMillis(6))
```

Comment:
Which thread are we blocking here? Is there any risk that it might be the instrumentation thread (the lifecycle-thread of the Activity-under-test)?

Reply:
Also: is `DelayHelper` doing something else that would be safer to use here perhaps?

Reply:
I have the quite opposite experience when it comes to displaying web content. I see the trend leaning towards being clear about when the user enters the web domain.

I myself think it's virtually impossible, in lack of a better superlative, to mimic native UI and, more importantly UX (behaviours, flows etc) in a web context. I think it's therefore better to be honest about it and put the expectations right from the beginning.

With that being said, I'd like to stress that I'm not asking for code changes here, but I think we should (re)assess our strategy when it comes to web content in native apps in general.

Reply:
Yea I guess this is dependant on what the user is expected to do. But overall, every app has integrated Chrome Custom Tabs https://developer.chrome.com/multidevice/android/customtabshttps://developer.chrome.com/multidevice/android/customtabs

Whatsapp, facebook, telegram, linkedin, reddit, medium just to name a few...

Main scenario for this was that it avoids the user leaving the app and having to maybe select a browser (therefore improving user retention if the user stays in the app) and also performance where a user may have a lot of previous tabs open in his browser and hence opening this fresh takes time to load.

In any case, we will review this as we add similar features.